### PR TITLE
Add Github action to run QGIS tests.

### LINKED ISSUE #231: CI for QGIS plugin
Currently there is none. We currently only manually check the plugin, see #202.

We should see if we can test at least a part of the plugin automatically. QGIS is available in Conda so perhaps that can be used, for instance to check if it loads properly.

Also Black and Ruff are only run on the python directory, not the qgis one. (EDIT: fixed)

### PR DESCRIPTION
**Pull request by evetion**
Fixes #231.

Relates to #320.

## Description
Use Docker Compose to setup a QGIS environment in which we can test our QGIS plugin.

Running this locally requires:

```bash
docker compose -f .docker/compose.yml up -d
docker exec -t qgis sh -c "qgis_setup.sh ribasim_qgis"
docker exec -t qgis sh -c "cd /tests_directory && qgis_testrunner.sh ribasim_qgis.tests"
```

Note that with the above, you can make changes to the test files, your local files are synced into the QGIS plugin directory in the image.

## Background
Few QGIS plugins have CI. Of the popular plugins that do are several by 3liz, I used https://github.com/3liz/QuickOSM for inspiration, without taking a dependency on their common toolchain https://github.com/3liz/qgis_plugin_tools. I hope to be as generic as possible, so this setup is easily shared by our other QGIS plugins. Any new tests named `qgis/tests/test_*.py`, will be run automatically, without needing to change the CI workflow (scripts).

### CODE REVIEW

**Comment by evetion** [comment]
Not sure what's happening with the codecov decreasing each time I make a PR.

**Comment by evetion** [comment]
```
Running test ribasim_qgis.tests.test_io ...
Warning: QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
ERROR: Status 2: File /tests_directory/ribasim_qgis.tests.test_io could not be found
QGIS Test Runner Inside - starting the tests ...
QGIS Test Runner - Trying to import ribasim_qgis.tests.test_io
QGIS Test Runner Inside - executing function <function run_all at 0x7feaf52a7640>
test_open_geopackage (ribasim_qgis.tests.test_io.TestIO) ... ok
test_passes (ribasim_qgis.tests.test_io.TestIO) ... ok

----------------------------------------------------------------------
Ran 2 tests in 0.001s

OK
Finished running test ribasim_qgis.tests.test_io (codes: IS_DEAD=1 IS_FAILED=1 IS_PASSED=0).
```

File: .docker/start.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
```
Comment (Hofer-Julian) [inline .docker/start.sh:1]:
```suggestion
#!/usr/bin/env bash
set -euxo pipefail
```

Please set the typical bash options: https://vaneyckt.io/posts/safer_bash_scripts_with_set_euxo_pipefail/

File: .docker/stop.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
```
Comment (Hofer-Julian) [inline .docker/stop.sh:1]: Same here

File: .docker/test.sh [hunk 3]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env bash
```
Comment (Hofer-Julian) [inline .docker/test.sh:1]: Same here

File: qgis/tests/test_io.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+from qgis.testing import unittest
+
+
+class TestIO(unittest.TestCase):
+    def test_passes(self):
+        self.assertTrue(True)
+
+    # TODO Open actual geopackage
+    def test_open_geopackage(self):
+        pass
```
Comment (Hofer-Julian) [inline qgis/tests/test_io.py:12]: I think we should wait with merging this PR until it actually tests something :)

Comment (evetion) [reply qgis/tests/test_io.py:12]: This actually does test that the plugin is installable and can load in QGIS.

Comment (Hofer-Julian) [reply qgis/tests/test_io.py:12]: I see.

Does that also work if you remove the dummy tests?

Comment (evetion) [reply qgis/tests/test_io.py:12]: And, besides being timeboxed, this does provide the quite generic boilerplate that our other X-qgis packages can use.

**Review by Hofer-Julian — COMMENTED** [review]
Two nitpicks, then it's fine to go from my side

**Review by Hofer-Julian — APPROVED** [review]

**Comment by evetion** [comment]
@JoerivanEngelen FYI

**Pull request merged**